### PR TITLE
`z_fireobj.h` header

### PR DESCRIPTION
**Pull request by AngheloAlf**
Move all the structs and functions related to `z_fireobj.c` file to its own header.
I also made some easy renames and cleanups

### CODE REVIEW

File: src/code/z_fireobj.c [hunk 1]
```diff
@@ -101,33 +95,35 @@ void FireObj_UpdateStateTransitions(PlayState* play, FireObj* fire) {
...
+    if ((fire->flags & FIRE_FLAG_1) && (fire->state != FIRE_STATE_NOT_LIT) &&
         WaterBox_GetSurface1_2(play, &play->colCtx, fire->position.x, fire->position.z, &waterY, &waterBox) &&
         (waterY - fire->position.y > 6500.0f * fire->yScale)) {
-        FireObj_SetState(fire, fire->dynamicSizeStep, FIRE_STATE_3);
+        FireObj_SetState(fire, fire->dynamicSizeStep, FIRE_STATE_NOT_LIT);
     }
-    if ((fire->flags & 2) && (player->heldItemAction == PLAYER_IA_STICK)) {
+    if ((fire->flags & FIRE_FLAG_2) && (player->heldItemAction == PLAYER_IA_STICK)) {
         Math_Vec3f_Diff(&player->meleeWeaponInfo[0].tip, &fire->position, &dist);
         if (Math3D_LengthSquared(&dist) < 400.0f) {
```
Comment (engineer124) [inline src/code/z_fireobj.c:118]:
```suggestion
        if (Math3D_LengthSquared(&dist) < SQ(20.0f)) {
```

File: src/code/z_fireobj.c [hunk 2]
```diff
@@ -101,33 +95,35 @@ void FireObj_UpdateStateTransitions(PlayState* play, FireObj* fire) {
...
+
     if (sp40) {
-        if (fire->state == FIRE_STATE_3) {
+        if (fire->state == FIRE_STATE_NOT_LIT) {
             if (player->unk_B28 > 0) {
-                FireObj_SetState(fire, fire->dynamicSizeStep, FIRE_STATE_0);
+                FireObj_SetState(fire, fire->dynamicSizeStep, FIRE_STATE_GROWING);
             }
         } else if (player->unk_B28 == 0) {
             player->unk_B28 = 0xD2;
```
Comment (engineer124) [inline src/code/z_fireobj.c:129]: `unk_B28` seems to be the deku stick lit/flame timer in this instance, so these can be dec (of course `unk_B28` seems to have other non-deku stick related purposes at a glance, classic player)

File: src/code/z_fireobj.c [hunk 3]
```diff
@@ -101,39 +95,41 @@ void FireObj_UpdateStateTransitions(PlayState* play, FireObj* fire) {
...
         }
         FireObj_SetState(fire, fire->dynamicSizeStep, nextState);
     }
-    if ((fire->flags & 1) && (fire->state != FIRE_STATE_3) &&
+
+    if ((fire->flags & FIRE_FLAG_1) && (fire->state != FIRE_STATE_NOT_LIT) &&
         WaterBox_GetSurface1_2(play, &play->colCtx, fire->position.x, fire->position.z, &waterY, &waterBox) &&
         (waterY - fire->position.y > 6500.0f * fire->yScale)) {
-        FireObj_SetState(fire, fire->dynamicSizeStep, FIRE_STATE_3);
+        FireObj_SetState(fire, fire->dynamicSizeStep, FIRE_STATE_NOT_LIT);
```
Comment (isghj5) [inline src/code/z_fireobj.c:114]: FIRE_FLAG_1 seems to be the only version that checks for water to be extinguishable, can we name it? Not sure what the second flag is tho.

Comment (AngheloAlf) [reply src/code/z_fireobj.c:114]: The second flag seems to be like "can interact with stick"
So what about `FIRE_FLAG_WATER_EXTINGUISHABLE` and `FIRE_FLAG_INTERACT_STICK`

Comment (engineer124) [reply src/code/z_fireobj.c:114]: Bump, this PR is good for me once these are named

Comment (AngheloAlf) [reply src/code/z_fireobj.c:114]: Whoops, completely forgot to check this PR again :sweat_smile:

**Review by engineer124 — APPROVED** [review]

**Review by hensldm — APPROVED** [review]

**Review by isghj5 — APPROVED** [review]

**Pull request merged**